common/persistence/elasticsearch/esVisibilityStore.go

```diff
@@ -793,3 +811,46 @@ func checkPageSize(request *p.ListWorkflowExecutionsRequestV2) {
 		request.PageSize = 1000
 	}
 }
+
+func processAllValuesForKey(dsl *fastjson.Value, keyFilter func(k string) bool,
+	processFunc func(obj *fastjson.Object, key string, v *fastjson.Value),
+) {
+	switch dsl.Type() {
+	case fastjson.TypeArray:
+		for _, val := range dsl.GetArray() {
+			processAllValuesForKey(val, keyFilter, processFunc)
+		}
+	case fastjson.TypeObject:
+		objectVal := dsl.GetObject()
+		objectVal.Visit(func(key []byte, val *fastjson.Value) {
+			keyString := string(key)
+			if keyFilter(keyString) {
+				processFunc(objectVal, keyString, val)
+			} else {
+				processAllValuesForKey(val, keyFilter, processFunc)
+			}
+		})
+	default:
+		// do nothing, since there's no key
+	}
+}
+
+func timeKeyFilter(key string) bool {
+	return timeKeys[key]
+}
+
+func timeProcessFunc(obj *fastjson.Object, key string, value *fastjson.Value) {
+	processAllValuesForKey(value, func(key string) bool {
+		return rangeKeys[key]
+	}, func(obj *fastjson.Object, key string, v *fastjson.Value) {
+		timeStr := string(v.GetStringBytes())
+
+		// try to parse time
+		parsedTime, err := time.Parse(defaultDateTimeFormat, timeStr)
+		if err == nil {
+			obj.Set(key, fastjson.MustParse(fmt.Sprintf(`"%v"`, parsedTime.UnixNano())))
+		}
+
+		// do nothing if failed to parse
```

Comment:
We probably shouldn't eat this error; caller should be able to return BadRequest when timeStr not parseable